### PR TITLE
policy/api: add more CRD validations

### PR DESCRIPTION
**Pull request by squeed**
Copying some logic from `Sanitize()` in to CRD validations:
- use the OpenAPI `cidr` format directly, remove baroque regex
- add OneOf for FQDN selector pattern vs. name
- add pre-existing MaxItems for port & ICMP rules
- add OneOf for L7 filter types

None of these add any new restrictions; they were always in the policy engine. Now these validation errors will be caught by the apiserver.

```release-note
More validation has been added to the CiliumNetworkPolicy and CiliumClusterwideNetworkPolicy CRDs. Policies that may have been ignored by the Cilium agent will now be rejected by the Kubernetes API server.
```

### CODE REVIEW

File: pkg/k8s/apis/cilium.io/client/crds/v2/ciliumclusterwidenetworkpolicies.yaml [hunk 1]
```diff
@@ -155,7 +156,7 @@ spec:
                       items:
                         description: 'CIDR specifies a block of IP addresses. Example:
                           192.0.2.1/32'
-                        pattern: ^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\/([0-9]|[1-2][0-9]|3[0-2])$|^s*((([0-9A-Fa-f]{1,4}:){7}(:|([0-9A-Fa-f]{1,4})))|(([0-9A-Fa-f]{1,4}:){6}:([0-9A-Fa-f]{1,4})?)|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){0,1}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){0,2}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){0,3}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){0,4}):([0-9A-Fa-f]{1,4})?))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){0,5}):([0-9A-Fa-f]{1,4})?))|(:(:|((:[0-9A-Fa-f]{1,4}){1,7}))))(%.+)?s*/([0-9]|[1-9][0-9]|1[0-1][0-9]|12[0-8])$
```
Comment (learnitall) [inline pkg/k8s/apis/cilium.io/client/crds/v2/ciliumclusterwidenetworkpolicies.yaml:158]: This is a fantastic clean up, thank you :partying_face:

**Review by learnitall — APPROVED** [review]
LGTM. Do you think it would make sense to add something to the upgrade notes? I'm assuming this wouldn't impact existing policies after an upgrade, but if folks are using a GitOps setup and reapplying resources related to these CRDs, they may run into new validation errors that weren't caught before, right?

**Comment by squeed** [comment]
> Do you think it would make sense to add something to the upgrade notes?

@learnitall not a bad idea, I'll update the release notes. Any policies caught by this would have been silently rejected before, which is arguably worse :-).

**Review by gandro — APPROVED** [review]

**Comment by squeed** [comment]
Holding until I add the release notes.

**Review by qmonnet — APPROVED** [review]

**Pull request merged**